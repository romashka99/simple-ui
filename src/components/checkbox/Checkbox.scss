@import '../../styles/colors.scss';
@import '../../styles/main.scss';
@import '../../styles/style.scss';

.checkbox {
	display: flex;
	align-items: center;
	flex-wrap: wrap;
	position: relative;
	cursor: pointer;

	.checkbox--label {
		padding-left: 8px;

		color: $color-grey;

		font-size: 16px;
		line-height: 24px;
	}

	.checkbox--inner {
		-webkit-appearance: none;
		box-sizing: border-box;
		outline: none;
		position: relative;
		margin: 0;

		cursor: pointer;

		width: 20px;
		height: 20px;

		background: $color-white;
		border: $border;
		border-radius: $border-radius;

		&:checked:after {
			content: url('data:image/svg+xml;utf8,<svg width="19" height="19" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M17.3565 5.49025C17.9273 4.87185 18.8914 4.83329 19.5098 5.40412C20.1281 5.97494 20.1667 6.93899 19.5959 7.55739L10.453 17.4621C9.866 18.0981 8.86781 18.118 8.25584 17.5061L4.44631 13.6965C3.85123 13.1015 3.85123 12.1366 4.44631 11.5416C5.0414 10.9465 6.00622 10.9465 6.6013 11.5416L9.28937 14.2296L17.3565 5.49025Z" /></svg>');
			position: absolute;
			top: 0;
		}

		&:disabled {
			cursor: not-allowed;
			background: $color-light-grey;
			border-color: $color-mid-light-grey;

			&:checked:after {
				content: url('data:image/svg+xml;utf8,<svg width="19" height="19" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M17.3565 5.49025C17.9273 4.87185 18.8914 4.83329 19.5098 5.40412C20.1281 5.97494 20.1667 6.93899 19.5959 7.55739L10.453 17.4621C9.866 18.0981 8.86781 18.118 8.25584 17.5061L4.44631 13.6965C3.85123 13.1015 3.85123 12.1366 4.44631 11.5416C5.0414 10.9465 6.00622 10.9465 6.6013 11.5416L9.28937 14.2296L17.3565 5.49025Z" /></svg>');
				position: absolute;
				top: 0;
			}

			+ .checkbox--label {
				color: $color-mid-light-grey;
			}
		}

		&:hover {
			&:not(:checked) {
				&:not(:disabled) {
				}
			}
		}
	}
}
