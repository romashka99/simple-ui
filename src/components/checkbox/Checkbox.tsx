import React, { DetailedHTMLProps, InputHTMLAttributes } from 'react';
import { Color, Size } from 'types/types';
import './Checkbox.scss';

interface ICheckboxProps
	extends DetailedHTMLProps<
			InputHTMLAttributes<HTMLInputElement>,
			HTMLInputElement
		>,
		React.AriaAttributes {
	color?: Color;
	scale?: Size;
	label?: string;
}

export default function Checkbox({
	color = 'default',
	scale = 'medium',
	label,
	className,
	...props
}: ICheckboxProps) {
	const classes = `checkbox checkbox--${color} checkbox--${scale}${
		className ? ' ' + className : ''
	}`;

	return (
		<label data-testid='checkbox' className={classes}>
			<input {...props} className='checkbox--inner' type='checkbox' />
			{label && <div className='checkbox--label'>{label}</div>}
		</label>
	);
}
