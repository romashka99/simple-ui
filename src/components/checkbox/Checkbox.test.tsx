import { render, screen } from '@testing-library/react';
import Checkbox from './index';

describe('Test Checkbox', () => {
	test('without attr', () => {
		render(<Checkbox />);
		const input = screen.getByTestId('checkbox');
		expect(input.classList.length).toBe(3);
		expect(input.classList.contains('checkbox')).toBe(true);
		expect(input.classList.contains('checkbox--medium')).toBe(true);
		expect(input.getAttribute('disabled')).toBeNull();
		expect(input).toMatchSnapshot();
	});
});
