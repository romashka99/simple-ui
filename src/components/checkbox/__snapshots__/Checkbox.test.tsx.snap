// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox checked 1`] = `
<label
  class="checkbox"
  data-testid="checkbox"
>
  <input
    class="checkbox--inner"
    data-testid="checkbox-inner"
    type="checkbox"
  />
</label>
`;

exports[`Test Checkbox disabled 1`] = `
<label
  class="checkbox"
  data-testid="checkbox"
>
  <input
    class="checkbox--inner"
    data-testid="checkbox-inner"
    disabled=""
    type="checkbox"
  />
</label>
`;

exports[`Test Checkbox has label 1`] = `
<label
  class="checkbox"
  data-testid="checkbox"
>
  <input
    class="checkbox--inner"
    data-testid="checkbox-inner"
    type="checkbox"
  />
  <span
    class="checkbox--label"
  >
    Label
  </span>
</label>
`;

exports[`Test Checkbox onChange 1`] = `
<label
  class="checkbox"
  data-testid="checkbox"
>
  <input
    class="checkbox--inner"
    data-testid="checkbox-inner"
    type="checkbox"
  />
</label>
`;

exports[`Test Checkbox without attr 1`] = `
<label
  class="checkbox"
  data-testid="checkbox"
>
  <input
    class="checkbox--inner"
    data-testid="checkbox-inner"
    type="checkbox"
  />
</label>
`;
